pkg/converter/v2/converter.go

```diff
@@ -252,12 +262,132 @@ func (c *converter) createChannels() error {
 }
 
-func extractID(value string) string {
-	title := strings.ToLower(value)
-	return strings.Join(strings.Split(title, " "), ".")
+func (c *converter) updateComponents() error {
+
+	components, ok := c.data["components"].(map[string]interface{})
+
+	if !ok {
+		return nil
+	}
+
+	parameters, ok := components["parameters"].(map[string]interface{})
+
+	if ok {
+		for _, rawParam := range parameters {
+			if param, ok := rawParam.(map[string]interface{}); ok {
+				delete(param, "name")
+				rawParam = param
+			}
+		}
+	}
+
+	messages, ok := components["messages"].(map[string]interface{})
+
+	if ok {
+		for _, messageRaw := range messages {
+			if message, ok := messageRaw.(map[string]interface{}); ok {
+				headersToSchema(message)
+			}
+
+		}
+	}
+
+	return nil
+}
+
+func alterParameters(parameters []interface{}, key string) (map[string]interface{}, error) {
+	paramsMap := make(map[string]interface{})
+	re := regexp.MustCompile(`{([^}]+)}`)
+	var paramNames []string
+
+	for _, part := range re.FindAll([]byte(key), -1) {
+		paramNames = append(paramNames, string(part))
+	}
+
+	for index, paramI := range parameters {
+
+		param, ok := paramI.(map[string]interface{})
+
+		if !ok {
+			return nil, asyncapierr.NewInvalidProperty("malformed parameter")
+		}
+
+		name := "default"
+
+		if paramName, ok := param["name"].(string); ok {
+			name = paramName
+		} else {
+			if len(paramNames) > index {
+				name = paramNames[index]
+			}
+		}
+		name = strings.TrimLeft(strings.TrimRight(name, "}"), "{")
+
+		if param["name"] != nil {
+			delete(param, "name")
+		}
+
+		paramsMap[name] = param
+
+	}
+	return paramsMap, nil
+}
+
+func (c *converter) alterChannels() error {
+	channels, ok := c.data["channels"].(map[string]interface{})
+
+	if !ok {
+		return asyncapierr.NewInvalidProperty("missing channels")
+	}
+
+	for key, item := range channels {
+		channel, ok := item.(map[string]interface{})
+		if !ok {
+			return asyncapierr.NewInvalidProperty("malformed channel")
+		}
+
+		if params, ok := channel["parameters"].([]interface{}); ok {
+
+			altered, err := alterParameters(params, key)
+			if err != nil {
+				return err
+			}
+			channel["parameters"] = altered
+		}
+
+		if publish, ok := channel["publish"].(map[string]interface{}); ok {
+			alterOperation(publish)
+		}
+
+		if subscribe, ok := channel["subscribe"].(map[string]interface{}); ok {
+			alterOperation(subscribe)
+		}
+	}
+	return nil
+}
+
+func headersToSchema(arg map[string]interface{}) {
+	if arg["headers"] != nil {
+		arg["headers"] = map[string]interface{}{
+			"type":       "object",
+			"properties": arg["headers"],
+		}
+	}
+}
+
+func alterOperation(operation map[string]interface{}) {
+	if message, ok := operation["message"].(map[string]interface{}); ok {
+		if oneOf, ok := message["oneOf"].([]map[string]interface{}); ok {
+			for _, elem := range oneOf {
+				headersToSchema(elem)
+			}
+		} else {
+			headersToSchema(message)
```

Comment:
I know that `alterOperation` and `headersToSchema` operate on indicator, but please return changed value from those function for better readable.